Bottom: 9d97a05d5690b67e0864484bcf3ffeead96929a1
Top:    e66cb5673f144e90a083d2cc0c9af86dc96c53b3
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 13:39:26 +0000

BORAST: Fixup to only draw the passed POLYAREA, not walk the linked list of POLYAREA and draw them all!


---

diff --git a/src/borast/borast-bentley-ottmann.c b/src/borast/borast-bentley-ottmann.c
index 7a0d1ac..762e2c0 100644
--- a/src/borast/borast-bentley-ottmann.c
+++ b/src/borast/borast-bentley-ottmann.c
@@ -1375,6 +1375,7 @@ poly_area_to_start_events (POLYAREA                *poly,
                                  counter, outer_contour);
         outer_contour = 0;
       }
+      break;
 
     } while ((pa = pa->f) != poly);
 
@@ -1401,6 +1402,7 @@ bo_poly_to_traps (hidGC gc, POLYAREA *poly, borast_traps_t *traps)
     for (contour = pa->contours; contour != NULL; contour = contour->next)
       num_events += contour->Count;
     /* FIXME: Remove horizontal edges? */
+    break;
   } while ((pa = pa->f) != poly);
 
   if (unlikely (0 == num_events))
